instrumentation/nginx/src/otel_ngx_module.cpp

```diff
@@ -328,6 +329,11 @@ void TraceContextCleanup(void* data) {
   context->~TraceContext();
 }
 
+void RequestContextMapCleanup(void* data) {
```

Comment:
Should this instead call the destructor? I'm not sure whether `clear` deallocates anything the map has allocated internally.

Reply:
@seemk Thank you for the review and for catching this. I have replaced the `clear()` call with the destructor.